phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java

```diff
@@ -502,25 +504,84 @@ private String getColumnInfoStringForTable(PTable table) {
         return getColumnInfoString(table, colInfo, columns, pkColumns);
     }
 
-    private boolean shouldContainQualifier(PTable table, List<PColumn> columns)
+    private boolean hasEncodedQualifier(PTable table)
     {
-        if (columns.size() == 0 || shouldGenerateWithDefaults) {
-            return false;
-        }
+        return table.getColumns().size() > 0
+                && !shouldGenerateWithDefaults
+                && table.getType() == PTableType.TABLE
+                && table.getEncodingScheme() != PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS;
+    }
 
-        return table.getType() == PTableType.TABLE && table.getEncodingScheme() != PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS;
+    private boolean isConsecutive(List<Integer> list) {
```

Comment:
We should also check that the first one is the normal starting value, otherwise this won't detect if the 1st column has been dropped.

Reply:
Now I see that you check that separately.
You may want to consolide those checks into this function.